### PR TITLE
Add deprecation message for postgresql resources and data sources

### PR DESCRIPTION
**Pull request by hcsa73**
Modified markdown description so it appears in the rendered docs

### CODE REVIEW

File: stackit/internal/services/postgresql/credential/datasource.go [hunk 1]
```diff
@@ -75,7 +76,16 @@ func (r *credentialDataSource) Configure(ctx context.Context, req datasource.Con
...
 func (r *credentialDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	descriptions := map[string]string{
-		"main":          "PostgreSQL credential data source schema. Must have a `region` specified in the provider configuration.",
+		"main": "PostgreSQL credential data source schema. Must have a `region` specified in the provider configuration.",
+		"deprecation_message": strings.Join(
+			[]string{
+				"The STACKIT PostgreSQL service will reach its end of support on June 30th.",
+				"Data sources of this type will stop work after that.",
+				"Use stackit_postgresqlflex_user instead.",
+				"For more details, check https://docs.stackit.cloud/stackit/en/bring-your-data-to-stackit-postgresql-flex-138347648.html",
```
Comment (joaopalet) [inline stackit/internal/services/postgresql/credential/datasource.go:85]:
```suggestion
				"The STACKIT PostgreSQL service will reach its end of support on June 30th 2024.",
				"Data sources of this type will stop work after that.",
				"Use `stackit_postgresqlflex_user` instead.",
				"For more details, check https://docs.stackit.cloud/stackit/en/bring-your-data-to-stackit-postgresql-flex-138347648.html",
```

**Review by joaopalet — DISMISSED** [review]

File: stackit/internal/services/postgresql/credential/datasource.go (hunk 1 above)
Comment (joaopalet) [reply stackit/internal/services/postgresql/credential/datasource.go:85]: Same suggestion for the other resources/data sources

File: stackit/internal/services/postgresql/instance/resource.go [hunk 2]
```diff
@@ -122,7 +122,16 @@ func (r *instanceResource) Configure(ctx context.Context, req resource.Configure
 // Schema defines the schema for the resource.
 func (r *instanceResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
 	descriptions := map[string]string{
-		"main":        "PostgreSQL instance resource schema. Must have a `region` specified in the provider configuration.",
+		"main": "PostgreSQL instance resource schema. Must have a `region` specified in the provider configuration.",
+		"deprecation_message": strings.Join(
+			[]string{
+				"The STACKIT PostgreSQL service will reach its end of support on June 30th 2024.",
+				"Resources of this type will stop working after that.",
+				"Use stackit_postgresqlflex_instance instead.",
```
Comment (joaopalet) [inline stackit/internal/services/postgresql/instance/resource.go:130]: Just noticed, the name of the resource is `stackit_postgresflex_...` and not `stackit_postgresqlflex_...`

**Review by joaopalet — CHANGES_REQUESTED** [review]

File: stackit/internal/services/postgresql/credential/resource.go [hunk 3]
```diff
@@ -98,7 +98,16 @@ func (r *credentialResource) Configure(ctx context.Context, req resource.Configu
 // Schema defines the schema for the resource.
 func (r *credentialResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
 	descriptions := map[string]string{
-		"main":          "PostgreSQL credential resource schema. Must have a `region` specified in the provider configuration.",
+		"main": "PostgreSQL credential resource schema. Must have a `region` specified in the provider configuration.",
+		"deprecation_message": strings.Join(
+			[]string{
+				"The STACKIT PostgreSQL service will reach its end of support on June 30th 2024.",
+				"Resources of this type will stop working after that.",
+				"Use stackit_postgresql_user instead.",
```
[the remark this answers is not in the document]
Comment (hcsa73) [reply stackit/internal/services/postgresql/credential/resource.go]: -.-' sorry

**Review by joaopalet — APPROVED** [review]

**Pull request merged**